### PR TITLE
langchain-community: Make ChatDatabricks model supports streaming response

### PR DESCRIPTION
**Pull request by WeichenXu123**
**Description:** Make ChatDatabricks model supports stream
**Issue:** N/A
**Dependencies:** MLflow nightly build version (we will release next MLflow version soon)
**Twitter handle:** N/A

Manually test:

(Before testing, please install `pip install git+https://github.com/mlflow/mlflow.git`)

```python
# Test Databricks Foundation LLM model
from langchain.chat_models import ChatDatabricks

chat_model = ChatDatabricks(
    endpoint="databricks-llama-2-70b-chat",
    max_tokens=500
)
from langchain_core.messages import AIMessageChunk

for chunk in chat_model.stream("What is mlflow?"):
  print(chunk.content, end="|")
```

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @baskaryan @case-k-git

File: libs/community/langchain_community/chat_models/mlflow.py [hunk 1]
```diff
@@ -119,9 +126,72 @@ def _generate(
...
+        **kwargs: Any,
+    ) -> Iterator[BaseMessageChunk]:
+        # We need to override `stream` to handle the case
+        # that `self._client` does not implement `predict_stream`
+        if not hasattr(self._client, "predict_stream"):
+            # MLflow deployment client does not implement streaming,
+            # so use default implementation
+            yield cast(
+                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
+            )
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/mlflow.py:160]: why not do this check in _stream?

Comment (WeichenXu123) [reply libs/community/langchain_community/chat_models/mlflow.py:160]: `checking it in `_stream` does not work.

When `chat_databricks_model.stream()` is invoked, if `ChatDatabricks` class does not override `stream` method but only override `_stream`, then this [line](https://github.com/WeichenXu123/langchain/blob/3218463f6a3d841905648971735949a14a16e191/libs/core/langchain_core/language_models/chat_models.py#L193) is executed, 

then `if type(self)._stream == BaseChatModel._stream` is always False, then it goes into this [branch](https://github.com/WeichenXu123/langchain/blob/3218463f6a3d841905648971735949a14a16e191/libs/core/langchain_core/language_models/chat_models.py#L199), i.e. it doesn't invoke `_stream`

Comment (WeichenXu123) [reply libs/community/langchain_community/chat_models/mlflow.py:160]: > why not do this check in _stream?

If so, before executing
```
yield cast(
                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
            )
```,

the `BaseChatModel.stream` will execute these lines code:

```
            config = ensure_config(config)
            messages = self._convert_input(input).to_messages()
            params = self._get_invocation_params(stop=stop, **kwargs)
            options = {"stop": stop, **kwargs}
            callback_manager = CallbackManager.configure(
                config.get("callbacks"),
                self.callbacks,
                self.verbose,
                config.get("tags"),
                self.tags,
                config.get("metadata"),
                self.metadata,
            )
            (run_manager,) = callback_manager.on_chat_model_start(
                dumpd(self),
                [messages],
                invocation_params=params,
                options=options,
                name=config.get("run_name"),
                run_id=config.pop("run_id", None),
                batch_size=1,
            )
```
this causes different behavior , 

I think if stream is not supported, `model.stream()` should only trigger 

```
            yield cast(
                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
            )
```
to keep the consistent behavir with master branch.

**Comment by WeichenXu123** [comment]
Gentle ping @baskaryan @case-k-git Would you take another look ? Thanks!

**Comment by WeichenXu123** [comment]
Gentle ping @baskaryan @case-k-git Would you take another look ? Thanks!

**Comment by WeichenXu123** [comment]
Gentle ping @baskaryan @case-k-git Would you take another look ? Thanks!

**Review by baskaryan — APPROVED** [review]

File: libs/community/langchain_community/chat_models/mlflow.py [hunk 2]
```diff
@@ -119,9 +123,76 @@ def _generate(
...
         return ChatMlflow._create_chat_result(resp)
 
+    def stream(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Iterator[BaseMessageChunk]:
```
Comment (WeichenXu123) [inline libs/community/langchain_community/chat_models/mlflow.py:151]: stream() interface.

**Pull request merged**